pkg/controllers/management/eks/eks_cluster_handler.go

```diff
@@ -304,14 +304,24 @@ func (e *eksOperatorController) onClusterChange(key string, cluster *mgmtv3.Clus
 			for key, value := range managedLaunchTemplateVersions {
 				managedLaunchTemplateVersionsToString[key] = value.(string)
 			}
-			cluster.DeepCopy()
+			cluster = cluster.DeepCopy()
 			cluster.Status.EKSStatus.ManagedLaunchTemplateVersions = managedLaunchTemplateVersionsToString
 			cluster, err = e.ClusterClient.Update(cluster)
 			if err != nil {
 				return cluster, err
 			}
 		}
 
+		generatedNodeRole, _ := status["generatedNodeRole"].(string)
+		if generatedNodeRole != "" && cluster.Status.EKSStatus.GeneratedNodeRole != generatedNodeRole {
+			cluster = cluster.DeepCopy()
+			cluster.Status.EKSStatus.GeneratedNodeRole = generatedNodeRole
+			cluster, err = e.ClusterClient.Update(cluster)
```

Comment:
I'm going to rubber stamp this because the rest of the file is like this, but this is another case where leveraging different controller framework paradigms (like `UpdateStatus`) may be more appropriate.